plugins/blueprint.cpp

```diff
@@ -789,8 +798,52 @@ command_result blueprint(color_ostream &out, vector<string> &parameters)
         end.z = -1;
 
     std::ostringstream err;
-    command_result result = do_transform(start, end, options, err);
-    if (result != CR_OK)
+    if (!do_transform(start, end, options, files, err))
+    {
         out.printerr("%s\n", err.str().c_str());
-    return result;
+        return false;
+    }
+    return true;
 }
+
+// entrypoint when called from Lua. returns the names of the generated files
+static int run(lua_State *L)
+{
+    int argc = lua_gettop(L);
+    vector<string> argv;
+
+    for (int i = 1; i <= argc; ++i)
+    {
+        const char *s = lua_tostring(L, i);
+        if (s == NULL)
+            luaL_error(L, "all parameters must be strings");
+        argv.push_back(s);
+    }
+
+    vector<string> files;
+    if (do_blueprint(Core::getInstance().getConsole(), argv, files))
```

Comment:
I think this output stream could use `Lua::GetOutput(L)` for the same reason. Note that this requires the Lua code to have been invoked with `SafeCall()`, and I wasn't able to easily trace the callers of `do_phase()` to figure out whether that's the case.

Reply:
Will update the code, but just to reply to your comment, there might not be any callers of the functions that call `do_phase()`. That function group is purely for backwards compatibility with the previous blueprint Lua API. There are no callers in-tree, but there might be someone somewhere who calls them. I was just keeping compatibility on principle.

Reply:
done

Reply:
That'd explain why I couldn't find any callers! Thanks for the update.